### PR TITLE
Audio API JSDoc

### PR DESCRIPTION
**Pull request by ctrlaltdavid**
JSDoc updated:
- Audio
- AudioEffectOptions
- SoundCache
- SoundObject

JSDoc added:
- AudioInjector

### CODE REVIEW

**Comment by ctrlaltdavid** [comment]
QA not required because no executable code changes.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11405/HighFidelity-Beta-PR13637-835fb92.apk

File: interface/src/scripting/Audio.h [hunk 1]
```diff
@@ -69,45 +78,91 @@ class Audio : public AudioScriptingInterface, protected ReadWriteLockable {
...
+     * var filename = File.getTempDir() + "/audio.wav";
+     * if (Audio.startRecording(filename)) {
+     *     Script.setTimeout(function () {
+     *         Audio.stopRecording();
+     *         print("Audio recording made in: " + filename);
+     *     }, 10000);
+     * 
+     * } else {
+     *     print("Could not make an audio recording in: " + filename);
+     * }
```
Comment (MiladNazeri) [inline interface/src/scripting/Audio.h]: may not be relevant, but can this audio recording be used in world, and if so, do we mention that anywhere?

File: libraries/audio/src/AudioEffectOptions.h [hunk 2]
```diff
@@ -16,35 +16,39 @@
...
- * @property {number} modRate=2.3
- * @property {number} modDepth=50
- * @property {number} earlyGain=0
- * @property {number} lateGain=0
- * @property {number} earlyMixLeft=20
- * @property {number} earlyMixRight=20
- * @property {number} lateMixLeft=90
- * @property {number} lateMixRight=90
- * @property {number} wetDryMix=50
+ * @property {number} bandwidth=10000 - The corner frequency (Hz) of the low-pass filter at reverb input.
```
Comment (MiladNazeri) [inline libraries/audio/src/AudioEffectOptions.h]: Just curious if this is the same as the .cpp file above and if so, does it need to be in both?

File: libraries/script-engine/src/AudioScriptingInterface.h [hunk 3]
```diff
@@ -35,91 +36,120 @@ class AudioScriptingInterface : public QObject, public Dependency {
...
+     * Triggered when the client is muted by the mixer because their loudness value for the noise background has reached the 
+     * threshold set for the domain in the server settings.
      * @function Audio.mutedByMixer
      * @returns {Signal} 
      */
     void mutedByMixer();
 
     /**jsdoc
-     * The entire environment has been muted by the mixer.
+     * Triggered when everyone is muted by the mixer.
```
Comment (MiladNazeri) [inline libraries/script-engine/src/AudioScriptingInterface.h]: when does this happen?  I think that part is a little unclear.

File: libraries/script-engine/src/AudioScriptingInterface.h [hunk 4]
```diff
@@ -35,91 +36,120 @@ class AudioScriptingInterface : public QObject, public Dependency {
...
     void noiseGateClosed();
 
     /**jsdoc
-     * A frame of mic input audio has been received and processed.
+     * Triggered when a frame of audio input is processed.
      * @function Audio.inputReceived
-     * @param {} inputSamples
+     * @param {Int16Array} inputSamples - The audio input processed.
      * @returns {Signal} 
      */
```
Comment (MiladNazeri) [inline libraries/script-engine/src/AudioScriptingInterface.h]: Can anything be done with the inputSample given back?

File: libraries/script-engine/src/ScriptAudioInjector.h [hunk 5]
```diff
@@ -26,19 +42,103 @@ class ScriptAudioInjector : public QObject {
...
+     * }, 1000);
+     */
     void finished();
 
 protected slots:
+
+    /**jsdoc
+     * Stop audio playback. (Synonym of {@link AudioInjector.stop|stop}.)
+     * @function AudioInjector.stopInjectorImmediately
+     */
```
Comment (MiladNazeri) [inline libraries/script-engine/src/ScriptAudioInjector.h]: Does this function have any advantage over just .stop?   not sure what the difference is if there is one.

File: interface/src/scripting/Audio.h (hunk 1 above)
Comment (ctrlaltdavid) [reply interface/src/scripting/Audio.h]: You could subsequently load the sound using `SoundChache.getSound()` and then use `Audio.playSound()` to play it. Both of these functions are documented. In the next commit, I've added, "Web, ATP, or file" to the description of the url in `SoundChache.getSound()`.

File: libraries/audio/src/AudioEffectOptions.h (hunk 2 above)
Comment (ctrlaltdavid) [reply libraries/audio/src/AudioEffectOptions.h]: Yes, unfortunately it needs to be in both: the same items are used both as properties of the AudioEffectOptions object and the type of the object optionally passed in when a new AudioEffecctOptions object is created.

File: libraries/script-engine/src/ScriptAudioInjector.h (hunk 5 above)
Comment (ctrlaltdavid) [reply libraries/script-engine/src/ScriptAudioInjector.h]: It's just a synonym, per the JSDoc. It could be deprecated and removed.

**Comment by huffman** [comment]
Looks good to me 👍 Pending Milad's approval

File: libraries/script-engine/src/AudioScriptingInterface.h (hunk 3 above)
Comment (ctrlaltdavid) [reply libraries/script-engine/src/AudioScriptingInterface.h]: JSDoc update.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11836/HighFidelity-Beta-PR13637-fdd6b7e.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11836/HighFidelity-Beta-PR13637-fdd6b7e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11836/HighFidelity-Beta-Interface-PR13637-fdd6b7e.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11836/HighFidelity-Beta-PR13637-fdd6b7e.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11836/HighFidelity-Beta-Interface-PR13637-fdd6b7e.exe

**Pull request merged**